presto-password-authenticators/src/main/java/com/facebook/presto/password/file/FileAuthenticator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.password.file;
+
+import com.facebook.airlift.http.server.BasicPrincipal;
+import com.facebook.presto.spi.security.AccessDeniedException;
+import com.facebook.presto.spi.security.PasswordAuthenticator;
+
+import javax.inject.Inject;
+
+import java.io.File;
+import java.security.Principal;
+import java.util.function.Supplier;
+
+import static com.google.common.base.Suppliers.memoizeWithExpiration;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class FileAuthenticator
+        implements PasswordAuthenticator
+{
+    private final Supplier<PasswordStore> passwordStoreSupplier;
+
+    @Inject
+    public FileAuthenticator(FileConfig config)
+    {
+        File file = config.getPasswordFile();
```

Comment:
Need a check to validate if file exists

Reply:
@ashishtadose I have added the validation

Reply:
looks good now